### PR TITLE
Activities not deleting

### LINKED ISSUE #474: [Activity]: Adding a new activity is not displayed when filters are applied.
Steps to Reproduce :

1. Login to the application
2. Click on Activities
3. Click on Add Activity
4. Enter the details to Add Activity
5. Apply the filters for the Activity Added
6. Notice, the Filters for the new added activity is not displayed

Note: When user manually searches for the new activity add, it is displayed and later when user applies the same filters as point 5 the filters output is displayed correctly.

Video : https://www.loom.com/share/759e38fed560444baff315459d21e6a7

### LINKED ISSUE #475: [Activities]: Unable to delete an activities
Steps to Reproduce
1. Login to the application
2. Click on Activities
3. Click  on Delete for any activity
4. Notice, the activity is not deleted.

Video : https://www.loom.com/share/879799eeee1a43dda09aa5fae1124fc2

### LINKED ISSUE #477: [Activity]: Edited Activity are not saved correctly
Steps to Reproduce
1. Login to the application
2. Click on Activities
3. Click on Edit button to edit an activity
4. Click on Save after Activity is edited.
5. Notice, the changes are not displayed

Note : When user signed out and Logins in again the edited changes are displayed on the activities page.
Video : https://www.loom.com/share/d8d4beb7d0404b21b8e0a4c189daf173

### PR DESCRIPTION
**Pull request by shailee-m**
closes #475 
closes #474 
closes #477  
Activity actions not updating reactively using props passed to child template

### CODE REVIEW

**Comment by brylie** [comment]
Is this ready for review, discarded, or still under work?

If still under work, lets try to avoid keeping state in the `Session`, if possible.

**Comment by shailee-m** [comment]
> Is this ready for review, discarded, or still under work?
> 
> If still under work, lets try to avoid keeping state in the `Session`, if possible.

It is ready to review. The problem here is the datatable library asks for templates here so there no way to provide listeners or callback methods to the action button templates in the table, that is why I have used sessions

**Comment by brylie** [comment]
OK, we can merge this as-is. Please resolve conflicts.

**Comment by shailee-m** [comment]
> OK, we can merge this as-is. Please resolve conflicts.

Done

**Comment by brylie** [comment]
It says this branch cannot be rebased due to conflicts.

**Comment by shailee-m** [comment]
> It says this branch cannot be rebased due to conflicts.

Rebased

File: app/client/views/activities/activities.js [hunk 1]
```diff
@@ -12,23 +12,36 @@ Template.activities.onCreated(function() {
...
   instance.totalRows = new ReactiveVar(0);
-  instance.reset = new ReactiveVar(true);
+
+  this.autorun(function() {
+    const activityDeleted = Session.get('activity-deleted');
+    const activityEdited = Session.get('activity-edited');
+    const currentPage = instance.currentPage.get();
+    const rowsPerPage = instance.rowsPerPage.get();
+    if (activityDeleted === true || activityEdited===true) {
+      onChange(instance, rowsPerPage, currentPage);
```
Comment (brylie) [inline app/client/views/activities/activities.js]: Just add some descriptive comments here, as this is the most important code. In particular, describe:

- what `onChange` is doing, as if you were teaching a junior developer or writing a tutorial
- why we needed to use `Session` here, as it is somewhat unconventional in this project

Comment (shailee-m) [reply app/client/views/activities/activities.js]: Done

**Pull request merged**